ios/ImagePickerManager.m

```diff
@@ -269,33 +299,126 @@ - (void)checkPermission:(void(^)(BOOL granted)) callback {
     else {
         if (@available(iOS 11.0, *)) {
             callback(YES);
-        } else {
+        }
+        else {
             [self checkPhotosPermissions:permissionBlock];
         }
     }
 }
 
-- (NSString*) getImageFileName:(NSString*)fileType {
+- (NSString *)getImageFileName:(NSString *)fileType
+{
     NSString *fileName = [[NSUUID UUID] UUIDString];
     fileName = [fileName stringByAppendingString:@"."];
     return [fileName stringByAppendingString:fileType];
 }
 
-+ (UIImage*)getUIImageFromInfo:(NSDictionary*)info {
++ (UIImage *)getUIImageFromInfo:(NSDictionary *)info
+{
     UIImage *image = info[UIImagePickerControllerEditedImage];
     if (!image) {
         image = info[UIImagePickerControllerOriginalImage];
     }
     return image;
 }
 
-+ (NSURL*)getNSURLFromInfo:(NSDictionary*)info {
++ (NSURL *)getNSURLFromInfo:(NSDictionary *)info {
     if (@available(iOS 11.0, *)) {
         return info[UIImagePickerControllerImageURL];
-    } else {
+    }
+    else {
         return info[UIImagePickerControllerReferenceURL];
     }
 }
 
 @end
 
+@implementation ImagePickerManager(PHPickerViewControllerDelegate)
+
+- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14))
+{
+    if (results.count == 0) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [picker dismissViewControllerAnimated:YES completion:^{
+                self.callback(@[@{@"didCancel": @YES}]);
+            }];
+        });
+        return;
+    }
+
+    // Immediately dismiss the controller to prevent calling the completion handler twice
+    [picker dismissViewControllerAnimated:YES completion:nil];
```

Comment:
@Johan-dutoit This change here fixes the double tap issue in my tests

Reply:
It does a better job, but still fails if you click really fast.

Reply:
You really want to try to break it 😅 I would suggest you make a call on how realistic this is (double tap speed vs. mouse clicks…). I don't see what other way there is to prevent this.

What's interesting is that it's reproducible (at least for me) only on the first double click. Subsequent attempts fail until you relaunch it. I'm able to reproduce this in a bare bones example and will report it as a bug to Apple.

```swift
class ViewController: UIViewController {
    @IBAction func pick(_ sender: Any) {
        let picker = PHPickerViewController(configuration: .init())
        picker.delegate = self
        present(picker, animated: true)
    }
}

extension ViewController: PHPickerViewControllerDelegate {
    func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
        picker.dismiss(animated: true)
        // Prints twice the first time you double click really quickly
        print("Called")
    }
}
```

Reply:
Reported to Apple as `FB9059292`

Reply:
I totally agree with your sentiments above.  At first I didn't even want to mention it, but decided that it's good to have a trail.

Reply:
Shouldn't the call to `dismissViewControllerAnimated` be made before we check for results? That's what's suggested [in the WWDC video about PHPicker](https://developer.apple.com/videos/play/wwdc2020-10652/?time=392), I'm wondering if that has incidence in the bug you're facing with double/triple taps.

Reply:
@flochtililoch Thanks for the comment. I made this change specifically to attempt to work around the mentioned bug. In my tests it didn't matter whether you call the completion block in the dismiss completion block or outside. It was always reproducible the _first_ time that you present the picker but not on the second call which to me suggest that there is an issue.

Reply:
Oh, what I meant was:

```objc
    // First, dismiss picker
    [picker dismissViewControllerAnimated:YES completion:nil];

    // Then, check results
    if (results.count == 0) {
        dispatch_async(dispatch_get_main_queue(), ^{
            [picker dismissViewControllerAnimated:YES completion:^{
                self.callback(@[@{@"didCancel": @YES}]);
            }];
        });
        return;
    }
    ...
```

Another thing worthwhile to explore, is around setting the flag `preferredAssetRepresentationMode` on the configuration object to `PHPickerConfigurationAssetRepresentationModeCurrent`. This [blog article](https://www.appcoda.com/phpicker/#crayon-60825b230a088198413804) suggests that not setting it can have a performance impact on the import process.

Reply:
Let's give this a go, would love to now face this :D

Reply:
I moved it to the first line of the delegate callback. The double tap problem persists but in terms of consistency I guess this is better (only one dismiss instead of separate handling for the "cancel" case).
I also applied the suggestion to set the `preferredAssetRepresentationMode`